pkg/agent/manager/sync.go

```diff
@@ -142,12 +142,22 @@ func (m *manager) updateEtriesSVIDs(entryRequestsList []*entryRequest, svids map
 				m.newClient([]string{ce.RegistrationEntry.SpiffeId}, ce.SVID, ce.PrivateKey)
 			}
 			m.cache.SetEntry(ce)
-			m.subscribers.Notify(ce)
+			sids := m.subscribers.Get(ce.RegistrationEntry.Selectors)
+			for _, id := range sids {
+				sub := m.subscribers.sidMap[id]
+				go m.notifySubscriber(sub)
+			}
```

Comment:
I liked the Notify method, maybe we could move this logic on it.

Reply:
The problem with having Notify method in subscribers is that we need to maintain a copy of cacheEntries in subscribers too which seems not too clean

Reply:
You are right. Then we'll leave it as is for now.